src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -451,9 +451,47 @@ public static ISet<INamespaceSymbol> GetUsingNamespacesInScope(this SemanticMode
                 }
             }
 
+            // For type constraints use declared accessibility of its member declaration (class, interface, delegate, method)
+            if (type.IsParentKind(SyntaxKind.TypeConstraint) &&
+                type.Parent.IsParentKind(SyntaxKind.TypeParameterConstraintClause))
```

Comment:
When can you have a TypeConstraint that is *not* parented by a TypeParameterConstraintClause?